extensions/wikia/Forum/ForumExternalController.class.php

```diff
@@ -58,6 +58,12 @@ public function createNewBoard() {
 			return;
 		}
 
+		try {
+			$this->displayAnyWriteAccessError();
```

Comment:
Two things.  First, I think functions like this should be named for what they intend to do, not what they might do in an exception.  In this case, that would be something like `checkEditToken()`.  Second, given a function that does this already exists as `checkWriteRequest()` why not call that here rather than define another function that just adds another layer of `try`/`catch` blocks?  If you want to reduce repetition, I'd make a method that just sets the return values for you and use it here, e.g.:
```php
function setMismatchError() {
    $this->status = 'error';
    $this->errormsg = wfMessage( 'forum-token-mismatch' )->escaped();
}
...
try {
    $this->checkWriteRequest();
} catch ( \BadRequestException $bre ) {
    $this->setMismatchError();
    return; # no need to return a boolean here; Oasis doesn't care what controllers return
}
```